x/nexus/keeper/wasm_message_route.go

```diff
@@ -0,0 +1,40 @@
+package keeper
+
+import (
+	"encoding/json"
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/axelarnetwork/axelar-core/x/nexus/exported"
+	types "github.com/axelarnetwork/axelar-core/x/nexus/types"
+)
+
+type req struct {
+	RouteMessages []exported.WasmMessage `json:"route_messages"`
+}
+
+// NewMessageRoute creates a new message route
+func NewMessageRoute(nexus types.Nexus, account types.AccountKeeper, wasm types.WasmKeeper) exported.MessageRoute {
+	return func(ctx sdk.Context, _ exported.RoutingContext, msg exported.GeneralMessage) error {
+		if msg.Asset != nil {
+			return fmt.Errorf("asset is not supported")
+		}
+
+		gateway := nexus.GetParams(ctx).Gateway
+		if gateway.Empty() {
+			return fmt.Errorf("gateway is not set")
```

Comment:
```suggestion
			return fmt.Errorf("wasm nexus gateway is not set")
```
or something like that